dot/network/README.md

```diff
@@ -0,0 +1,178 @@
+# Gossamer `network` Package
+
+This package implements the [peer-to-peer networking capabilities](https://crates.parity.io/sc_network/index.html)
+provided by the [Substrate](https://docs.substrate.io/) framework for blockchain development, which implies that it is
+built on the extensible [`libp2p` networking stack](https://docs.libp2p.io/introduction/what-is-libp2p/). `libp2p`
+provides implementations of a number of battle-tested peer-to-peer (P2P) networking protocols (e.g. [Noise](#noise) for
+[key exchange](#identities--key-management), and [Yamux](#yamux) for [stream multiplexing](#stream-multiplexing)), and
+also makes it possible to implement the blockchain-specific protocols defined by Substrate (e.g. [syncing](#sync) and
+[finalising](#GRANDPA) blocks, and maintaining the [transaction pool](#transactions)). The purpose of this document is
+to provide the information that is needed to understand the P2P networking capabilities that are implemented by
+Gossamer - this includes an introduction to P2P networks and `libp2p`, as well as detailed descriptions of the Gossamer
+P2P networking protocols.
+
+## Peer-to-Peer Networking & `libp2p`
+
+[Peer-to-peer](https://en.wikipedia.org/wiki/Peer-to-peer) networking has been a dynamic field of research for over two
+decades, and P2P protocols are at the heart of many blockchain networks. P2P networks can be contrasted with traditional
```

Comment:
```suggestion
decades, and P2P protocols are at the heart of blockchain networks. P2P networks can be contrasted with traditional
```

i hope all blockchains use p2p haha, maybe this isn'true? would be concerning if not